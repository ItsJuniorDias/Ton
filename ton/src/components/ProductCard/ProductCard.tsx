import React, { useRef } from 'react';
import { View, Text } from 'react-native';
import { Button } from '../../components';
import { formatCurrency } from '../../utils';

import {
  Container,
  Thumbnail,
  Title,
  Price,
  PriceDivided,
  Installments,
} from './styles';

type ProductCard = {
  id: string;
  title: string;
  thumbnail: string;
  price: number;
};

export const ProductCard = ({ id, title, thumbnail, price }: ProductCard) => {
  const testIDs = useRef({
    container: 'containerCard_testID',
  }).current;

  return (
    <Container testID={testIDs.container} key={id}>
      <Thumbnail source={{ uri: thumbnail }} resizeMode="contain" />

      <Title numberOfLines={1}>{title}</Title>

      <View>
        <Price>{formatCurrency(price)}</Price>

        <Installments>
          <Installments>12x </Installments>
          <PriceDivided>{formatCurrency(price / 12)}</PriceDivided>
        </Installments>
      </View>

      <Button title="Pedir agora" />
    </Container>
  );
};
