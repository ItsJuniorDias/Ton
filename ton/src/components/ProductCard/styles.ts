import styled from 'styled-components/native';
import { scale } from '../../utils';

export const Container = styled.View`
  width: 46%;
  height: ${scale(228)};
  background-color: ${({ theme }) => theme.colors.shape};
  margin-top: ${scale(24)};
  margin-right: ${scale(24)};
  border-radius: ${scale(16)};
  padding: ${scale(16)};
`;

export const Thumbnail = styled.Image`
  height: ${scale(75)};
`;

export const Title = styled.Text`
  font-family: ${({ theme }) => theme.fonts.regular};
  font-size: ${scale(16)};
  color: ${({ theme }) => theme.colors.text};
  text-align: center;
  margin-top: ${scale(16)};
`;

export const Price = styled.Text`
  font-family: ${({ theme }) => theme.fonts.bold};
  font-size: ${scale(12)};
  color: ${({ theme }) => theme.colors.title};
  line-height: ${scale(24)};
`;

export const Installment = styled.View`
  flex-direction: row;
  align-items: center;
`;

export const Installments = styled.Text`
  color: ${({ theme }) => theme.colors.title};
  font-size: ${scale(11)};
  font-family: ${({ theme }) => theme.fonts.bold};
`;

export const PriceDivided = styled.Text`
  color: ${({ theme }) => theme.colors.primary};
  font-size: ${scale(18)};
  font-family: ${({ theme }) => theme.fonts.bold};
`;
