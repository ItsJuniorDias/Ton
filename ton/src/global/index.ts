export * from './shadowStyles';
export * from './theme';
