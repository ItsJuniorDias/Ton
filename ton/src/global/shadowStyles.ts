export const shadowStyles = {
  shadowColor: '#000',
  shadowOffset: {
    width: 0,
    height: 5,
  },
  shadowOpacity: 0.1,
  shadowRadius: 2.62,
  elevation: 2,
};
