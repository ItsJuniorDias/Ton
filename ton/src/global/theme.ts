export const theme = {
  colors: {
    primary: '#01ad0b',
    secondary: '#01ea34',
    shape: '#ffffff',
    background: '#eff4f8',
    title: '#000000',
    text: '#20252b',
    notification: '#ff7b00',
    alert: '#ed5e69',
  },
  fonts: {
    regular: 'Roboto-Regular',
    medium: 'Roboto-Medium',
    bold: 'Roboto-Bold',
  },
};
