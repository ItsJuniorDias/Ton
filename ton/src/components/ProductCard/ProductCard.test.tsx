import 'jest-styled-components';
import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { ProductCard } from './ProductCard';
import { theme } from '../../global';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');

jest.mock('react-native-iphone-x-helper', () => ({
  getStatusBarHeight: jest.fn(),
  isIphoneX: jest.fn(),
}));

describe('Behavior ProductCard', () => {
  const props = {
    id: 'T1',
    title: 'T1 Brother',
    thumbnail:
      'https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-1',
    price: 118.8,
  };

  const screenRender = () => (
    <ThemeProvider theme={theme}>
      <ProductCard {...props} />
    </ThemeProvider>
  );

  it('render snapshot', () => {
    const result = render(screenRender()).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('should render ProductCard', () => {
    const { getByTestId } = render(screenRender());

    const containerCard = getByTestId('containerCard_testID');

    expect(containerCard).toBeTruthy();
  });
});
