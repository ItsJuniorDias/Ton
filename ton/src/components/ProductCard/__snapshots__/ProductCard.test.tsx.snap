// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior ProductCard render snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 31,
        "height": 447,
        "marginRight": 47,
        "marginTop": 47,
        "paddingBottom": 31,
        "paddingLeft": 31,
        "paddingRight": 31,
        "paddingTop": 31,
        "width": "46%",
      },
    ]
  }
  testID="containerCard_testID"
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "https://res.cloudinary.com/dunz5zfpt/fl_progressive/f_auto,c_limit,w_256,q_auto/site-ton/t1-1",
      }
    }
    style={
      [
        {
          "height": 147,
        },
      ]
    }
  />
  <Text
    numberOfLines={1}
    style={
      [
        {
          "color": "#20252b",
          "fontFamily": "Roboto-Regular",
          "fontSize": 31,
          "marginTop": 31,
          "textAlign": "center",
        },
      ]
    }
  >
    T1 Brother
  </Text>
  <View>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Roboto-Bold",
            "fontSize": 24,
            "lineHeight": 47,
          },
        ]
      }
    >
      R$ 118,80
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Roboto-Bold",
            "fontSize": 22,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "Roboto-Bold",
              "fontSize": 22,
            },
          ]
        }
      >
        12x 
      </Text>
      <Text
        style={
          [
            {
              "color": "#01ad0b",
              "fontFamily": "Roboto-Bold",
              "fontSize": 35,
            },
          ]
        }
      >
        R$ 9,90
      </Text>
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "#01ea34",
          "borderRadius": 31,
          "height": 51,
          "justifyContent": "center",
          "marginTop": 31,
          "width": "100%",
        },
        {
          "opacity": 1,
        },
      ]
    }
    testID="button_testID"
  >
    <Text
      style={
        [
          {
            "color": "#20252b",
            "fontFamily": "Roboto-Bold",
            "fontSize": 27,
            "textAlign": "center",
          },
        ]
      }
    >
      Pedir agora
    </Text>
  </View>
</View>
`;
